Mathlib/RingTheory/Polynomial/Basic.lean

```diff
@@ -460,6 +467,30 @@ theorem ker_modByMonicHom (hq : q.Monic) :
 
 end ModByMonic
 
+/-- Let `P` be a polynomial over `R`. If its constant term is a unit and its other coefficients are
+nilpotent, then `P` is a unit. -/
+theorem IsUnit.isUnit_of_isNilpotent {P : Polynomial R} (hunit : IsUnit (P.coeff 0))
```

Comment:
Sorry I missed that on first read. I think this name is a bit too generic (there's no way to guess it's about polynomials!), I would suggest simply `Polynomial.isUnit_of_coeff_isUnit_isNilpotent`, or if you really prefer dot notation something like `IsUnit.isUnit_polynomial_of_isNilpotent` just to reference polynomials.

Reply:
I thought that due to the namespace `Polynomial` (opened on line 43), the result would automatically be named `Polynomial.IsUnit.isUnit_of_isNilpotent`. Am I misunderstanding namespaces? Is the namespace `Polynomial` closed before `IsUnit.isUnit_of_isNilpotent`?

Reply:
Oh sorry I missed that :facepalm: I would still suggest using `isUnit_of_coeff_isUnit_isNilpotent` since the dot notation won't trigger anyway with your current name, but the most important is that it's in namespace `Polynomial` so you can keep the current name if you prefer.

Reply:
I changed the name to `isUnit_of_coeff_isUnit_isNilpotent`. Thank you!